### PR TITLE
fix(attack-path): highlight the causal chain when a finding is picked from a drawer/summary list (#7187)

### PR DESCRIPTION
**Pull request by Seb-MIGUEL**
## Summary

- Clicking a finding directly on the attack-path graph already highlighted its causal chain correctly, but picking one from a drawer list or a summary card (e.g. one of several "Captured Files") did not: no path lit up, even though the detail panel opened correctly.
- Root cause: a finding's type cluster on its endpoint collapses into a "+N" node once it has more than 4 findings of that type. The drawer/card click seeded the highlight walk on the finding's raw backend id, which has no node of its own while the cluster is collapsed - scopeChainFlowToSeeds then fell back to showing the entire unscoped chain, with nothing to highlight (its seed still didn't match anything rendered).
- buildCausalChainFlow now also returns causalSourceByFinding, the map it already built internally from a finding id to whatever node actually represents it (itself once expanded, its type cluster while collapsed, the "+N" endpoint cluster when its whole endpoint is overflow-hidden). SimulationAttackPath resolves a selected finding through this map before seeding the chain scope and the highlight walk, so it always seeds on something that is actually rendered - reproduced and confirmed against a live simulation's real graph data before fixing.

## Test plan

- [x] Added a causalSourceByFinding unit test covering both the collapsed (resolves to the type cluster) and expanded (resolves to itself) cases.
- [x] Added a unit test for the third resolution case: findings on overflow-hidden endpoints resolve to their "+N" endpoint cluster (review follow-up).
- [x] Added a scopeChainFlowToSeeds unit test seeding on the resolved id, confirming it scopes correctly instead of hitting the "whole chain" fallback; per review feedback it asserts the resolved mapping is defined before seeding, so a mapping regression fails loudly.
- [x] Full attack_path test suite passes (101/101).
- [x] tsc --noEmit and eslint clean.

### CODE REVIEW

File: openaev-front/src/__tests__/admin/components/simulations/simulation/attack_path/attack-path-flow-helpers.test.ts [hunk 1]
```diff
@@ -1291,6 +1298,20 @@ describe('scopeChainFlowToSeeds', () => {
...
+  it('scopes down correctly, instead of falling back to the full chain, when the seed is resolved through causalSourceByFinding first', () => {
+    // Reproduces picking one of the 5 collapsed findings from a drawer/summary list (not clicking an
+    // already-rendered graph node): the caller must resolve the raw finding id to whatever actually
+    // represents it (its cluster, here) before seeding, exactly as SimulationAttackPath's
+    // effectiveSelectedFindingId does — seeding on the raw id instead is the previous bug, covered by
+    // the "falls back to the full chain" test above.
+    const chainFlow = buildCausalChainFlow(collapsed, tt);
+    const resolvedSeed = chainFlow.causalSourceByFinding.get('NODE_FINDING|share|a');
+
+    const scoped = scopeChainFlowToSeeds(chainFlow, new Set([resolvedSeed as string]));
```
[the remark this answers is not in the document]
Comment (SamuelHassine) [reply openaev-front/src/__tests__/admin/components/simulations/simulation/attack_path/attack-path-flow-helpers.test.ts]: Done in f780e2f: the test now asserts expect(resolvedSeed).toBeDefined() first and seeds with resolvedSeed! instead of the as-string cast, so a causalSourceByFinding regression fails loudly on the mapping assertion itself rather than as a confusing downstream scoping mismatch. Also added a sibling test covering the third resolution case (findings on overflow-hidden endpoints resolving to their +N endpoint cluster).

**Review by SamuelHassine — APPROVED** [review]
Full review done: read buildCausalChainFlow and scopeChainFlowToSeeds in full plus the changed regions of SimulationAttackPath.tsx with surrounding context. The fix is correct and well-scoped: causalSourceByFinding only ever maps to nodes that are actually placed in the flow (itself, its type cluster, or the overflow endpoint cluster), the memo dependency arrays are correct, and there is no stale-closure risk since the resolution is memo-derived and the click handlers only write state. Addressed the Copilot test-cast comment and added coverage for the hidden-endpoint resolution case. Lint, tsc and the attack_path suite (101/101) verified locally; CI fully green.

**Comment by SamuelHassine** [comment]
Review-and-fix pass complete. Summary of what was done:

- Independent review of the full changed files (buildCausalChainFlow, scopeChainFlowToSeeds, and the changed regions of SimulationAttackPath.tsx with surrounding context): the resolution correctly handles all three representation cases (finding itself, collapsed type cluster, overflow-hidden endpoint cluster - every map value is a placed node, and scopeChainFlowToSeeds keeps its presentSeedIds fallback as a safety net), memo dependency arrays are correct, and there is no stale-closure risk (resolution is memo-derived; click handlers only write state).
- Addressed the Copilot review comment on the test: the resolved seed is now asserted with expect(resolvedSeed).toBeDefined() and seeded via a non-null assertion instead of an "as string" cast, so a causalSourceByFinding regression fails loudly on the mapping itself. Thread replied to and resolved.
- Added a unit test for the previously uncovered third resolution branch: findings on overflow-hidden endpoints (more than 4 distinct endpoints at a depth) resolve to their "+N" endpoint cluster.
- Verified locally in a clean worktree: eslint clean, tsc --noEmit clean, attack_path vitest suite 101/101.
- CI fully green on the final head, including both required checks (quality-gates Quality Status, Artifact & Image Sanity Check) and all codecov patch/project checks.
- PR body updated to reflect the final state; title already matches the repo convention.

Remaining before merge: the requested human review from @guillaumejparis (reviewDecision is still REVIEW_REQUIRED; code-owner review is enforced on main).
